src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -5180,31 +5180,190 @@ int wm_vuldet_collect_agent_software(scan_agent *agent, sqlite3 *db, scan_ctx_t*
     return retval;
 }
 
-void *wm_vuldet_main(wm_vuldet_t * vuldet) {
-    wm_vuldet_init(vuldet);
+void wm_vuldet_create_and_send_message(vu_task_context_t task, task_status TASK_STATUS, char* details) {
+    wm_vuldet_task_msg_ctx* message = NULL;
+    char status_message[OS_SIZE_128] = {0};
 
-    wm_vuldet_check_db();
+    if (task.taskID != (size_t)OS_INVALID) {
+        os_calloc(1, sizeof(wm_vuldet_task_msg_ctx), message);
+        message->task_id = task.taskID;
+        message->task_status = TASK_STATUS;
+
+        switch (TASK_STATUS) {
+            case WM_TASK_IN_PROGRESS:
+                snprintf(status_message, OS_SIZE_128, "Task in progress.%s", details ? details : "");
+                os_strdup(status_message, message->status_message);
+                break;
+            case WM_TASK_CANCELLED:
+                snprintf(status_message, OS_SIZE_128, "Task cancelled by the user.%s", details ? details : "");
```

Comment:
ditto

Reply:
I was trying to avoid an unnecessary space when  `details` is NULL. 
Now the call to `wm_vuldet_create_and_send_message()` has a space in the beginning of `details`.